parser/parser/include/ruby_parser/driver.hh

```diff
@@ -207,6 +207,7 @@ class max_numparam_stack {
         ruby_parser::node_list *decls;
         bool staticContext = false;
         token_t opening_token = nullptr;
+        bool hasItParam = false; // Track if 'it' parameter is used in this scope
```

Comment:
Instead of adding a new bool (one per stack item), you can smuggle this into the `max` with a special value like `std::numeric_limits<int>::max` (and adjust the checks accordingly)